### PR TITLE
feat: add support for 'tildeop' option (~ as operator)

### LINKED ISSUE #3094: Support `tildeop`
Vim allows tilde `~` to behave as an operator (http://vimdoc.sourceforge.net/htmldoc/options.html#'tildeop') and it would be nice if VSCodeVim supported this behavior as well.

### PR DESCRIPTION
**Pull request by benjaAguilar**
add support for the `tildeop` option, allowing `~` to behave as an operator when enabled, matching Vim behavior.

- When `tildeop = false` (default):
  - `~` toggles the case of a single character (unchanged behavior)

- When `tildeop = true`:
  - `~` acts as an operator and requires a motion (e.g. `~w`, `~$`)
  - `~~` toggles the entire line
  - `g~` continues to work as before

- In Visual and Visual Line modes:
  - `~` always toggles the selected text (independent of `tildeop`)

## Implementation details

- Added `vim.tildeop` configuration option
- Updated action resolution to conditionally treat `~` as:
  - command
  - operator
  - motion (when used after an operator)
- Ensured compatibility with existing operator flow (`g~`)
- Preserved undo/redo behavior

## Tests

Added  tests covering:

- Default behavior (`tildeop = false`)
- Operator behavior (`~w`, `~$`, `2~w`)
- Edge cases (`~~`, `g~~`)
- Visual and Visual Line modes
- Dynamic toggling via `:set tildeop` / `:set notildeop`

This implementation Closes #3094

### CODE REVIEW

File: src/actions/commands/actions.ts [hunk 1]
```diff
@@ -1001,6 +1001,23 @@ class ToggleCaseAndMoveForward extends BaseCommand {
     return newText;
   }
 
+  public override doesActionApply(vimState: VimState, keysPressed: string[]): boolean {
+    const isVisual =
+      vimState.currentMode === Mode.Visual ||
+      vimState.currentMode === Mode.VisualLine ||
+      vimState.currentMode === Mode.VisualBlock;
```
Comment (J-Fields) [inline src/actions/commands/actions.ts]: Replace with [`isVisualMode()`](https://github.com/VSCodeVim/Vim/blob/master/src/mode/mode.ts#L50)

File: src/actions/operator.ts [hunk 2]
```diff
@@ -436,6 +438,46 @@ class ToggleCaseOperator extends ChangeCaseOperator {
     }
     return newText;
   }
+
+  public override doesActionApply(vimState: VimState, keysPressed: string[]): boolean {
+    const isVisual =
+      vimState.currentMode === Mode.Visual ||
+      vimState.currentMode === Mode.VisualLine ||
+      vimState.currentMode === Mode.VisualBlock;
```
Comment (J-Fields) [inline src/actions/operator.ts]: Replace with [`isVisualMode()`](https://github.com/VSCodeVim/Vim/blob/master/src/mode/mode.ts#L50)

**Comment by benjaAguilar** [comment]
Applied the requested changes and re-ran the tests successfully. Thanks for the review!

File: src/actions/commands/actions.ts [hunk 3]
```diff
@@ -1001,6 +1002,18 @@ class ToggleCaseAndMoveForward extends BaseCommand {
     return newText;
   }
 
+  public override doesActionApply(vimState: VimState, keysPressed: string[]): boolean {
+    if (isVisualMode(vimState.currentMode)) {
+      return super.doesActionApply(vimState, keysPressed);
+    }
+
+    if (configuration.tildeop) {
```
Comment (J-Fields) [inline src/actions/commands/actions.ts]:
```suggestion
    if (configuration.tildeop && !isVisualMode(vimState.currentMode)) {
```

File: src/actions/operator.ts [hunk 4]
```diff
@@ -424,6 +425,7 @@ class LowerCaseWithMotion extends LowerCaseOperator {
 @RegisterAction
 class ToggleCaseOperator extends ChangeCaseOperator {
   public keys = [['g', '~'], ['~']];
+  override createsUndoPoint = true;
```
Comment (J-Fields) [inline src/actions/operator.ts]:
```suggestion
```

Should be unnecessary; it's already true in the base class

File: src/actions/operator.ts [hunk 5]
```diff
@@ -436,6 +438,41 @@ class ToggleCaseOperator extends ChangeCaseOperator {
...
+      return configuration.tildeop;
+    }
+
+    return super.doesActionApply(vimState, keysPressed);
+  }
+}
+
+// Add motion support for '~' in operator-pending context
+@RegisterAction
+class TildeMotion extends BaseMovement {
```
Comment (J-Fields) [inline src/actions/operator.ts]: Is this necessary? See `BaseOperator.runRepeat` and its call in `ModeHandler`.

File: package.json [hunk 6]
```diff
@@ -451,6 +451,10 @@
       "title": "Vim",
       "type": "object",
       "properties": {
+        "vim.tildeop": {
+          "type": "boolean",
```
Comment (J-Fields) [inline package.json:455]: Let's add `markdownDescription`

File: src/configuration/configuration.ts
Comment (J-Fields) [inline src/configuration/configuration.ts:36]: Let's add the `top` alias

**Review by J-Fields — APPROVED** [review]
Thanks!
